File: app/src/main/java/com/example/test/UserDetailFragment.java
package com.example.test;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.FragmentActivity;
import androidx.lifecycle.ViewModelProvider;

import com.example.test.data.details.UserDetail;
import com.example.test.databinding.FragmentUserDetailBinding;
import com.example.test.model.fragment.ViewBindingFragment;
import com.example.test.viewmodels.DetailViewModel;

import dagger.hilt.android.AndroidEntryPoint;
import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers;
import io.reactivex.rxjava3.core.Single;
import io.reactivex.rxjava3.disposables.CompositeDisposable;
import io.reactivex.rxjava3.disposables.Disposable;
import io.reactivex.rxjava3.schedulers.Schedulers;

@AndroidEntryPoint
public class UserDetailFragment extends ViewBindingFragment<FragmentUserDetailBinding> {

    private DetailViewModel viewModel;
    private String login;
    private final CompositeDisposable disposables = new CompositeDisposable();

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
            UserDetailFragmentArgs args = UserDetailFragmentArgs.fromBundle(getArguments());
            login = args.getLogin();
        }
    }

    @NonNull
    @Override
    protected FragmentUserDetailBinding inflateBinding(@NonNull LayoutInflater inflater, @Nullable ViewGroup container) {
        return FragmentUserDetailBinding.inflate(inflater, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        getBinding().setClickListener(v -> {
            FragmentActivity activity = getActivity();
            if (activity != null) {
                activity.onBackPressed();
            }
        });

        viewModel = new ViewModelProvider(this).get(DetailViewModel.class);
        subscribeUi(login);
    }

    private void subscribeUi(String login) {
        Single<UserDetail> single = viewModel.getResult(login);
        Disposable disposable = single.subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(
                        userDetail -> getBinding().setDetail(userDetail),
                        throwable -> {
                            Toast.makeText(requireContext(), "Error loading user details", Toast.LENGTH_SHORT).show();
                        }
                );
        disposables.add(disposable);
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        disposables.clear();
    }
}

File: app/src/main/java/com/example/test/UserDetailFragment.kt
package com.example.test

import android.os.Bundle
import android.text.Html
import android.text.method.LinkMovementMethod
import android.text.util.Linkify
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.core.text.HtmlCompat
import androidx.core.view.isVisible
import androidx.fragment.app.Fragment
import androidx.fragment.app.viewModels
import androidx.lifecycle.lifecycleScope
import androidx.navigation.fragment.navArgs
import com.bumptech.glide.Glide
import com.example.test.databinding.FragmentDetailViewBinding
import com.example.test.viewmodels.DetailViewModel
import dagger.hilt.android.AndroidEntryPoint
import kotlinx.coroutines.Job
import kotlinx.coroutines.launch

@AndroidEntryPoint
class UserDetailFragment : Fragment() {

    private val args: UserDetailFragmentArgs by navArgs()
    private var binding: FragmentDetailViewBinding? = null
    private val viewModel: DetailViewModel by viewModels()
    private var searchJob: Job? = null

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        binding = FragmentDetailViewBinding.inflate(inflater, container, false)
        return binding!!.root
    }

    override fun onDestroyView() {
        searchJob?.cancel()
        binding = null
        super.onDestroyView()
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        binding!!.ivClose.setOnClickListener {
            fragmentManager?.popBackStack()
        }
        viewModel.userDetail.observe(viewLifecycleOwner) {
            binding?.apply {
                tvName.text = it.name
                tvBio.text = it.bio
                tvLogin.text = it.login
                tvAdmin.isVisible = it.site_admin
                tvLocation.text = it.location
                setUrlSpan(it.blog)
                Glide.with(view.context)
                    .load(it.avatar_url)
                    .circleCrop()
                    .placeholder(R.drawable.github_b_24px)
                    .into(ivAvatar)
            }
        }
        subscribeUi(args.login)
    }

    private fun setUrlSpan(url: String?) {
        binding?.apply {
            url?.let {
                tvBlog.movementMethod = LinkMovementMethod.getInstance()
                tvBlog.autoLinkMask = Linkify.WEB_URLS
                tvBlog.text = Html.fromHtml(url, HtmlCompat.FROM_HTML_MODE_COMPACT)
            } ?: kotlin.run {
                ivLink.isVisible = false
                tvBlog.isVisible = false
            }
        }
    }

    private fun subscribeUi(login: String) {
        searchJob?.cancel()
        searchJob = lifecycleScope.launch {
            val detailData = viewModel.getResult(login)
            viewModel.userDetail.postValue(detailData)
        }
    }
}